### DISCUSSION TITLE
sqlite backend how to decode _content column?  access the database parse raw _content?

### DISCUSSION THREAD

**Discussion by Lexonight1** [reclosedev/requests-cache#955] [Q&A] [answered]
it took about a day to create my cache and now i am just using sqlite to queery the data in the database. what is _content encoded as? gzip?

**Comment by Lexonight1** [comment, marked as answer]
nevermind, i would just think it would be nice to keeps backups of cahces database and access them without request_cache hitting to url again in case it was expired
```
connection = sqlite3.connect(db_file)
cur = connection.cursor()
x = cur.execute('Select value from responses').fetchone()
code  = json.loads(x[0])
blob_data = base64.b85decode(code['_content'])
print(type(blob_data))
print(blob_data.decode('UTF-8'))
```

**Comment by JWCook** [comment]
# In Python
If you want to access cached responses in python, use `CachedSession.cache.responses` or `.filter()`.

Example:
```py
from requests_cache import CachedSession

session = CachedSession()
r = session.get('https://httpbin.org/get')
for response in session.cache.responses.values():
    print(response.content)
```
See the [cache inspection](https://requests-cache.readthedocs.io/en/stable/user_guide/inspection.html) section of the docs for more details.

# In SQL
For querying the SQLite db directly (with only SQL), a few extra steps are needed:
1. Use the [JSON serializer](https://requests-cache.readthedocs.io/en/stable/user_guide/serializers.html) with `decode_content=True`.
    * This is mainly used for the Filesystem backend, to provide human-readable JSON files, but will also work with SQLite.
3. Use SQLite's [json_extract()](https://www.sqlite.org/json1.html#jmini) function on the `_decoded_content` key
```py
session = CachedSession('http_cache_json.sqlite', serializer='json', decode_content=True)
r = session.get('https://httpbin.org/get')
```

From a `sqlite3` prompt:
```sql
SELECT json_extract(value, '$._decoded_content') FROM responses;
```

Or from the command line:
```sh
 sqlite3 http_cache_json.sqlite 'SELECT json_extract(value, \'$._decoded_content\') FROM responses;'
```

# Converting serialization formats
See [Exporting to a different backend](https://requests-cache.readthedocs.io/en/stable/user_guide/backends.html#exporting-to-a-different-backend) if you want to convert your existing SQLite cache to this format.

**Reply by Lexonight1** [reply]
ty appreciate it, yeah it seams i didn't set that to begin with, but can still access the data thanks again
